services/src/main/java/org/keycloak/services/resources/ThemeResource.java

```diff
@@ -107,7 +108,9 @@ public Response getResource(@PathParam("version") String version, @PathParam("th
     @Produces(MediaType.APPLICATION_JSON)
     public Response getLocalizationTexts(@PathParam("realm") String realmName, @QueryParam("theme") String theme,
                                          @PathParam("locale") String localeString, @PathParam("themeType") String themeType,
-                                         @QueryParam("source") boolean showSource) throws IOException {
+                                         @QueryParam("source") boolean showSource, @QueryParam("hasWords") List<String> hasWords,
+                                         @QueryParam("first") @DefaultValue("0") long first,
+                                         @QueryParam("max") long max) throws IOException {
```

Comment:
Please use `Integer` instead of `long`, and account for `null` in the code.

```suggestion
                                         @QueryParam("first") Integer first,
                                         @QueryParam("max") Integer max) throws IOException {
```

Reply:
the signature of skip is `long` though